src/Npgsql/TypeHandler.cs

```diff
@@ -156,6 +156,14 @@ internal virtual object ReadPsvAsObject(DataRowMessage row, FieldDescription fie
 
         public virtual bool PreferTextWrite { get { return false; } }
 
+        public virtual Type EnumType
```

Comment:
I think it would be better not to have this on TypeHandler... In the develop branch the IEnumHandler interface (implemented by all concrete EnumHandlers) already exposes an EnumType property. We can expose this same property in the 3.0.x branch, and then you can simply check if a TypeHandler implements the interface. For arrays of enums you'd have to add the additional check for array handler and then do the same interface check on its element handler.

What do you think?